common/src/main/java/com/microsoft/identity/common/internal/platform/AsymmetricKeyFactory.java

```diff
@@ -0,0 +1,59 @@
+//  Copyright (c) Microsoft Corporation.
+//  All rights reserved.
+//
+//  This code is licensed under the MIT License.
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files(the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions :
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+package com.microsoft.identity.common.internal.platform;
+
+import com.microsoft.identity.common.exception.ClientException;
+
+/**
+ * Interface defining a factory for asymmetric keys.
+ */
+public interface AsymmetricKeyFactory {
+
+    /**
+     * Generates a new {@link AsymmetricKey}, retrievable by the provided alias.
```

Comment:
Javadoc basic expectations:
- 2048 bit keys (secure & widely compatible on Android devices)
- Maybe hardware backed (depending on OS ver, TPM/HSM chipset)
- API 18+ assumed, API 23+ is best security

Reply:
0556f65a1fb7d137b5214aee9d6d8c3da49c46cb